subsys/net/ip/ipv6.c

```diff
@@ -2131,6 +2131,9 @@ int net_ipv6_send_ns(struct net_if *iface,
 
 		if (net_is_ipv6_addr_unspecified(&NET_IPV6_HDR(pkt)->src)) {
 			NET_DBG("No source address for NS");
+			if (pending) {
+				net_pkt_unref(pending);
+			}
 			goto drop;
```

Comment:
In zephyr we try to leave empty line after "if" block, so please put empty line before the goto statement.